src/test/java/de/tum/in/www1/artemis/service/GradingScaleServiceTest.java

```diff
@@ -130,11 +126,9 @@ void testMatchPercentageToGradeStepWithBonusPoints(double bonusPercentage) {
     void testSaveGradingScaleInvalidGradeStepsNoGradeName() {
         createCustomGradeStep("", 70, 95);
 
-        BadRequestAlertException exception = assertThrows(BadRequestAlertException.class, () -> gradingScaleService.saveGradingScale(gradingScale));
-
-        assertThat(exception.getEntityName()).isEqualTo("gradeStep");
-        assertThat(exception.getErrorKey()).isEqualTo("invalidGradeStepFormat");
-        assertThat(exception.getMessage()).isEqualTo("Not all grade steps are following the correct format.");
+        assertThatExceptionOfType(BadRequestAlertException.class).isThrownBy(() -> gradingScaleService.saveGradingScale(gradingScale))
+                .withMessage("Not all grade steps are following the correct format.")
+                .matches(exception -> exception.getEntityName().equals("gradeStep") && exception.getErrorKey().equals("invalidGradeStepFormat"));
```

Comment:
```suggestion
                .matches(exception -> "gradeStep".equals(exception.getEntityName()) && "invalidGradeStepFormat".equals(exception.getErrorKey()));
```
I'll leave it here. Please check the other `.matches(...)` occurrences yourself.

Reply:
I saw this recommended by codacity. Is it just to prevent possible NullPointer exceptions, or is there also any other reason for this? 
I fixed it in all occurences

Reply:
It's about nullpointers indeed.